projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, Input, OnInit, ViewEncapsulation } from '@angular/core';
+import { FilterConfig, FilterSetting } from '../../models/filter.model';
+import { FilterService } from '../../services/filter/filter.service';
+
+@Component({
+  selector: 'xuilib-generic-filter',
+  templateUrl: 'generic-filter.component.html',
+  styleUrls: ['generic-filter.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class GenericFilterComponent implements OnInit {
+
+  @Input() public config: FilterConfig;
+  @Input() public settings?: FilterSetting;
+
+  constructor(private readonly filterService: FilterService) { }
+
+  ngOnInit() {
+    if (!this.settings) {
+      this.getSettings();
+    }
+  }
+
+  public applyFilter() {
+    this.filterService.persist(this.settings, this.config.persistence);
+  }
+
+  public cancelFilter() {
+    this.getSettings();
+  }
+
+  public isSelected(id: string, key: string): boolean {
+    if (this.settings) {
+      return this.settings.fields.some(x=> x.name == id) && this.settings.fields.some(x => x.value.some(x => x == key));
```

Comment:
I am not sure about this logic. using some inside some.

Reply:
its just determining if the settings exist for that key. what would you prefer i use, indexof?

Reply:
My question was does it work. Do you have a test for it?